src/ciphersuite/errors.rs

```diff
@@ -9,6 +9,11 @@ pub(crate) enum HKDFError {
     InvalidLength,
 }
 
+#[derive(Debug, PartialEq)]
+pub(crate) enum HpkeError {
```

Comment:
Please use `implement_error`. With #274 landed the others in here will use it already.

Reply:
But the question is whether we need a separate error here. I'm not too happy with the errors in here anyway. HPKE and evercrypt have their own errors that we get in here. The question is now how we want to expose them to the rest of the library. We could simply forward them (not great because they don't follow the same error handling logic). So maybe the best thing would be to have something like a `CryptoError` that holds the underlying non-openmls error values or just their error message.

Reply:
Maybe also keep in mind that we want to abstract the whole crypto layer some day, so we shouldn't propagate errors from other libraries.